os/arch/arm/src/amebad/amebad_serial.c

```diff
@@ -808,7 +786,16 @@ int up_lowgetc(void)
  *  sent character
  *
  ****************************************************************************/
-int up_putc(int ch)
+
+#define MAX_BUFFER_SIZE 1000
+extern bool abort_mode;
```

Comment:
I think this is not good option, @sunghan-chang Is there any api or something to realize abort situation? or should we add this to dev structure? let us here your opinion.

Reply:
I agree @Taejun-Kwon 's opinion. But before this, I don't know why up_putc should be different with abort mode. Please let me know it first.

Reply:
When assert fails during application context, both printf and lldbg will output at the same time. So by disabling uart interrupts during this time, we can prevent mixed logs. As abort mode is set to true after the failure of debug assert, I used it.

Reply:
Do you mean that printf for threads is working in assert mode? If we add controlling the interrupt always, what's the problem? The stm has it in up_putc.

Reply:
True. both stm and imxrt have disabled interrupts in up_putc while outputting each character. During assert failure, we won't wait for \n, instead, we print the logs directly (This is what I implemented) so that in the future even if we miss \n in assert failure logs the logs won't be stuck in the buffer. I also have tried disabling uart interrupts while outputting each character but it didn't resolve the issue (still mixed logs occurred). So in order to prevent mixed logs we need to disable the uart interrupts till the whole line is output.